ls1intum/ArTEMiS
# Feature/system notifications

### Checklist

### Motivation and Context
Students, Tutors, and Instructors were not notified about changes in the system, so a notification system should notify them.

### Description
This Pull Request adds the administration of system notification and notifications for exercise creations and updates. For quiz exercises, each state is sent to the client.

### Steps for Testing
1. Log in to ArTEMiS
2. Navigate to Notification Management
3. Create a new system notification, with notification date now and expire date in the future
4. Notification should be shown at the top of the container
5. create a quiz exercise
6. a notification should appear for a new exercise
7. make quiz visible and start quiz
8. two notifications should appear in the notification container

### Screenshots
![image](https://user-images.githubusercontent.com/10998002/55239015-1b39df80-5236-11e9-9833-01f5fb8456a0.png)
![image](https://user-images.githubusercontent.com/10998002/55239057-2ee54600-5236-11e9-9354-cd449cffbd14.png)
